### PR TITLE
[CALCITE-6400] MAP_ENTRIES function should throw if a key value is null

### PR DESCRIPTION
**Pull request by caicancai**
https://issues.apache.org/jira/browse/CALCITE-6400

### CODE REVIEW

**Comment by caicancai** [comment]
In spark, this is an error that occurs in function execution, not in type verification, so I modified the method in sqlfunctions

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java [hunk 1]
```diff
@@ -7735,23 +7735,16 @@ void checkRegexpExtract(SqlOperatorFixture f0, FunctionAlias functionAlias) {
     final SqlOperatorFixture f = f0.withLibrary(SqlLibrary.SPARK);
     f.checkScalar("map_entries(map['foo', 1, 'bar', 2])", "[{foo, 1}, {bar, 2}]",
         "RecordType(CHAR(3) NOT NULL f0, INTEGER NOT NULL f1) NOT NULL ARRAY NOT NULL");
-    f.checkScalar("map_entries(map['foo', 1, null, 2])", "[{foo, 1}, {null, 2}]",
-        "RecordType(CHAR(3) f0, INTEGER NOT NULL f1) NOT NULL ARRAY NOT NULL");
+
```
[the remark this answers is not in the document]
Comment (caicancai) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:7738]: @chucheng92 I looked at the git submission record and it was chucheng92 who submitted these tests. Can you help me check this PR when you are free?

[the remark this answers is not in the document]
Comment (chucheng92) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:7738]: thanks for reporting this issue, I will take a look.

[the remark this answers is not in the document]
Comment (chucheng92) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:7738]: These lines are the element cast scenarios I copied and added based on the previous author, so if map_entries keys don't support null key, these cases can be safely deleted.

**Comment by chucheng92** [comment]
looks good to me in general. just one question, does this exception have to be thrown at runtime? i noticed that the exception is thrown directly in SqlFunctions.

**Comment by caicancai** [comment]
> looks good to me in general. just one question, does this exception have to be thrown at runtime? i noticed that the exception is thrown directly in SqlFunctions.

@mihaibudiu Thank you for your review. I write this because spark has this mechanism, but calcite is usually checked in Opera and Checker. Which method do you think should be used?

**Comment by mihaibudiu** [comment]
I don't know what "opera" and "checker" are.

**Review by mihaibudiu — APPROVED** [review]

**Comment by caicancai** [comment]
> I don't know what "opera" and "checker" are.

Sorry, typo, it should be OperaandChecker

**Comment by chucheng92** [comment]
@caicancai could you rename the current commit name? It is different from jira and pr names.

**Review by chucheng92 — APPROVED** [review]

**Comment by caicancai** [comment]
@mihaibudiu @chucheng92  Good to merge？

**Comment by mihaibudiu** [comment]
The issue is about map_entries, but this PR fixes the arguments to the map constructor.

**Comment by caicancai** [comment]
> The issue is about map_entries, but this PR fixes the arguments to the map constructor.

Thanks, I will modify the jira summary, the description is being discussed in jira

**Comment by caicancai** [comment]
@mihaibudiu Can it be merged?

**Pull request merged**